### ISSUE TITLE
AweCsomeUser.GetSiteUserById failing since last update

### ISSUE THREAD

**Issue by taylanken** [OleAlbers/aweCsome#24]
SharePoint-Platform: O365
VS-Version: 2017
AweCsome Version: 2.0.2-CI-20190813.4

Since yesterday's update to latest AweCsome version __2.0.2-CI-20190813.4__, the following code fails:

```csharp
public static AweCsomeUser GetUser(ClientContext clientContext, int id)
{
    string hostUrl = Query.GetHostUrl();
    if (!People.ContainsKey(hostUrl)) People.Add(hostUrl, new List<AweCsomeUser>());
    var existingUser = People[hostUrl].FirstOrDefault(q => q.Id == id);
    if (existingUser == null)
    {
        IAweCsomePeople aweCsomePeople = new AweCsomePeople(clientContext);
        existingUser= aweCsomePeople.GetSiteUserById(id);
        People[hostUrl].Add(existingUser);
    }
    return existingUser;
}
```

More specifically, the line `existingUser= aweCsomePeople.GetSiteUserById(id);` fails with following error message:

> Microsoft.SharePoint.Client.CollectionNotInitializedException
  HResult=0x80131509
  Message=The collection has not been initialized. It has not been requested or the request has not been executed. It may need to be explicitly requested.
  Source=Microsoft.SharePoint.Client.Runtime
  StackTrace:
   at Microsoft.SharePoint.Client.ClientObjectCollection`1.<GetEnumerator>d__0.MoveNext()
   at AweCsome.AweCsomePeople.ToAweCsomeUser(User user, Boolean getGroups)
   at AweCsome.AweCsomePeople.AweCsome.Interfaces.IAweCsomePeople.GetSiteUserById(Int32 id)
   at LHIND.Mydea.Logic.Helpers.SharePoint.GetUser(ClientContext clientContext, Int32 id) in C:\Users\u529783\Source\Repos\Mydea\MydeaEntityO365\Logic\Helpers\SharePoint.cs:line 43
   at LHIND.Mydea.Logic.Helpers.SharePoint.GetUserName(ClientContext clientContext, Int32 id) in C:\Users\u529783\Source\Repos\Mydea\MydeaEntityO365\Logic\Helpers\SharePoint.cs:line 57
   at LHIND.Mydea.Logic.DatabaseLogic.FillMissingUsers[T](T entity) in C:\Users\u529783\Source\Repos\Mydea\MydeaEntityO365\Logic\DatabaseLogic.cs:line 82
   at LHIND.Mydea.Logic.DatabaseLogic.Insert[T](T entity) in C:\Users\u529783\Source\Repos\Mydea\MydeaEntityO365\Logic\DatabaseLogic.cs:line 108
   at LHIND.Mydea.Logic.StructureLogic.CreateGroups() in C:\Users\u529783\Source\Repos\Mydea\MydeaEntityO365\Logic\StructureLogic.cs:line 83
   at LHIND.Mydea.Logic.StructureLogic.CreateGenericLists() in C:\Users\u529783\Source\Repos\Mydea\MydeaEntityO365\Logic\StructureLogic.cs:line 68
   at LHIND.Mydea.Logic.StructureLogic.Install() in C:\Users\u529783\Source\Repos\Mydea\MydeaEntityO365\Logic\StructureLogic.cs:line 175
   at LHIND.Mydea.Web.Controllers.StructureController.Install() in C:\Users\u529783\Source\Repos\Mydea\MydeaEntityO365\MydeaEntityO365Web\Controllers\StructureController.cs:line 66
   at System.Web.Mvc.ActionMethodDispatcher.Execute(ControllerBase controller, Object[] parameters)
   at System.Web.Mvc.ReflectedActionDescriptor.Execute(ControllerContext controllerContext, IDictionary`2 parameters)
   at System.Web.Mvc.ControllerActionInvoker.InvokeActionMethod(ControllerContext controllerContext, ActionDescriptor actionDescriptor, IDictionary`2 parameters)
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.<>c.<BeginInvokeSynchronousActionMethod>b__9_0(IAsyncResult asyncResult, ActionInvocation innerInvokeState)
   at System.Web.Mvc.Async.AsyncResultWrapper.WrappedAsyncResult`2.CallEndDelegate(IAsyncResult asyncResult)
   at System.Web.Mvc.Async.AsyncResultWrapper.WrappedAsyncResultBase`1.End()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.EndInvokeActionMethod(IAsyncResult asyncResult)
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.AsyncInvocationWithFilters.<>c__DisplayClass11_0.<InvokeActionMethodFilterAsynchronouslyRecursive>b__0()
   at System.Web.Mvc.Async.AsyncControllerActionInvoker.AsyncInvocationWithFilters.<>c__DisplayClass11_2.<InvokeActionMethodFilterAsynchronouslyRecursive>b__2()

**Comment by OleAlbers** [comment]
yikes! Will take a look

**Comment by OleAlbers** [comment]
[FIXED] in 2.0.3-CI-20190814.1
(Could take an hour until available in NuGet)
